### DISCUSSION TITLE
How does Visualize handle changed Shared Dimensions on already published graphics

### DISCUSSION THREAD

**Discussion by tboeni** [visualize-admin/visualization-tool#1232] [General]
Hi everyone,
This a question I was asking myself lately. When I create a Shared Dimension for something that I know will be replaced by a proper SD in due time, how would Visualize handle this trasition. Does it change the for already published graphics too, as for example in a CMS document?

**Comment by bprusinowski** [comment]
Hi @tboeni, I think it depends on how big is the change to the shared dimension and whether it's used in the chart of not.

- If it's not used anywhere in the chart, it shouldn't cause problems.
- If it's used as a filter, and its IRI or the IRI of value used in a filter change, then the chart would be broken (probably showing no data).
- If it's mapped to a field, and its IRI change, then the chart would be broken due to a dimension that can't be found.
- If it's mapped to a field, and its IRI doesn't change, then the chart should still work (depending on whether its values changed, and whether they were used as multi-filters).
- If it's used as e.g. segmentation and its value IRIs change, the chart could be broken (assuming some filters were applied).

I probably missed some cases, and the severity of the chart being broken might also differ (e.g. it might still display some data if e.g. some value IRIs changed, that had custom color mapping). Generally however, I think that any substantial change to a shared dimension used in a chart will most probably break the chart.

Let me know if this helps, or if you have more questions :)
